snapcraft/plugins/rust.py

```diff
@@ -102,6 +104,39 @@ def build(self):
             cmd.append(' '.join(self.options.rust_features))
         self.run(cmd, env=self._build_env())
 
+    def _write_cross_compile_config(self):
+        if not self.project.is_cross_compiling:
+            return
+
+        # Cf. http://doc.crates.io/config.html
+        os.makedirs(self._cargo_dir, exist_ok=True)
+        with open(os.path.join(self._cargo_dir, 'config'), 'w') as f:
+            f.write('''
+                [build]
+                target = "{}"
+
+                [target.{}]
+                linker = "{}"
+                '''.format(self._target, self._target,
+                           '{}-gcc'.format(self.project.arch_triplet)))
+
+    def enable_cross_compilation(self):
+        # Cf. rustc --print target-list
+        rust_targets = {
+            'armhf': 'armv7-{}-{}eabihf',
+            'arm64': 'aarch64-{}-{}',
+            'i386': 'i686-{}-{}',
+            'amd64': 'x86_64-{}-{}',
+            'ppc64el': 'powerpc64le-{}-{}',
+        }
+        self._target = rust_targets.get(self.project.deb_arch).format(
+            'unknown-linux', 'gnu')
+        if not self._target:
```

Comment:
If there isn't an entry in the above dict for `self.project.deb_arch`, you'll never get here-- the use of `format` will cause `'NoneType' object has no attribute 'format'`. If you get rid of the format this will work, though.